services/codecov/codecov.tester.js

```diff
@@ -1,23 +1,59 @@
 'use strict'
 
-const { ServiceTester } = require('../tester')
 const { isIntegerPercentage } = require('../test-validators')
-
-const t = (module.exports = new ServiceTester({
-  id: 'codecov',
-  title: 'Codecov.io',
-}))
+const t = (module.exports = require('../tester').createServiceTester())
 
 t.create('gets coverage status')
-  .get('/c/github/codecov/example-python.json')
+  .get('/github/codecov/example-python.json')
   .expectBadge({
     label: 'coverage',
     message: isIntegerPercentage,
   })
 
-t.create('gets coverate status for branch')
-  .get('/c/github/codecov/example-python/master.json')
+t.create('gets coverage status for branch')
+  .get('/github/codecov/example-python/master.json')
   .expectBadge({
     label: 'coverage',
     message: isIntegerPercentage,
   })
+
+t.create('handles unknown repository')
+  .get('/github/codecov2/fake-not-even-a-little-bit-real-python.json')
+  .expectBadge({
+    label: 'coverage',
+    message: 'repository not found',
+  })
+
+t.create('handles unauthorized error')
+  .get('/gh/codecov/private-example-python.json')
+  .intercept(nock =>
+    nock('https://codecov.io/api')
+      .get('/gh/codecov/private-example-python')
+      .reply(401)
+  )
+  .expectBadge({
+    label: 'coverage',
+    message: 'not authorized to access repository',
+  })
```

Comment:
This test is arguably helpful to ensure that things are wired up correctly, though it feels like it's mostly a test of `errorMessages`.

Could it use a real private project instead?

Reply:
Fair point! I don't actually know of any private projects that have Codecov enabled, but if anyone has such a repo and is willing to share I'd be happy to plug it in.

Reply:
I don't know one… Maybe we could just leave a comment.

Reply:
Works for me!